frappe/permissions.py

```diff
@@ -333,14 +333,14 @@ def check_user_permission_on_link_fields(d):
 				# restricted for this link field, and no matching values found
 				# make the right message and exit
 				if d.get("parentfield"):
-					# "Not allowed for Company = Restricted Company in Row 3. Restricted field: reference_type"
-					msg = _("Not allowed for {0}: {1} in Row {2}. Restricted field: {3}").format(
-						_(field.options), d.get(field.fieldname), d.idx, field.fieldname
+					# "You are not allowed to access Company 'Restricted Company' in row 3, field Reference Type"
+					msg = _("You are not allowed to access {0} '{1}' in row {2}, field {3}").format(
+						_(field.options), d.get(field.fieldname) or _("empty"), d.idx, _(field.label)
 					)
 				else:
-					# "Not allowed for Company = Restricted Company. Restricted field: reference_type"
-					msg = _("Not allowed for {0}: {1}. Restricted field: {2}").format(
-						_(field.options), d.get(field.fieldname), field.fieldname
+					# "You are not allowed to access Company 'Restricted Company' in field Reference Type"
+					msg = _("You are not allowed to access {0} '{1}' in field {2}").format(
```

Comment:
this change will invalidate translations for all the languages for this string. maybe revert it? or we can also change it in all the translation files (?)

(same for the above)

Reply:
True. Unfortunately I do not speak 62 languages. I just take care of german users and some of them are rightfully confused when they see the current message.

Maybe we should improve the translation workflow in general so it's not the responsibility of devs anymore?

Reply:
Nevermind, forgot we have usable AI now